### PR TITLE
RDM-10041 Work Allocation BaseTypes

### PR DESCRIPTION
**Pull request by RebeccaBaker**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-10041

### Change description ###
Added three new base types for work Allocation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: repository/src/main/resources/db/changelog/db.changelog-rdm-10041.xml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                        and jurisdiction_id is null
+                        and version = (select max(version) from field_type
+                        where reference = 'Complex'
+                        and jurisdiction_id is null
+                        and base_field_type_id is null))" />
+        </insert>
+
+        <insert tableName="complex_field">
+            <column name="reference" value="baseLocation" />
+            <column name="label" value="Organisation To Add" />
```
Comment (danlysiak) [inline repository/src/main/resources/db/changelog/db.changelog-rdm-10041.xml:87]: Needs updating!

File: repository/src/main/resources/db/changelog/db.changelog-rdm-10041.xml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                        and jurisdiction_id is null)" />
+            <column name="complex_field_type_id"
+                    valueComputed="(select id from field_type where reference = 'CaseLocation'
+                        and version = 1
+                        and jurisdiction_id is null)" />
+        </insert>
+
+        <insert tableName="complex_field">
+            <column name="reference" value="region" />
+            <column name="label" value="Organisation To Remove" />
```
Comment (danlysiak) [inline repository/src/main/resources/db/changelog/db.changelog-rdm-10041.xml:102]: Needs updating!

File: repository/src/main/resources/db/changelog/db.changelog-rdm-10041.xml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet id="rdm-10041" author="rebecca.baker@hmcts.net">
+        <!-- FixedList40 -->
+        <insert tableName="field_type">
+            <column name="created_at" value="${now}" />
+            <column name="reference" value="FixedList40" />
+            <column name="version" value="1" />
+            <column name="maximum" value="40" />
+        </insert>
```
Comment (danlysiak) [inline repository/src/main/resources/db/changelog/db.changelog-rdm-10041.xml:12]: Although this is in the requirements, creating a new base type seems like adding more complexity, if it is simply a known base type with constraints.. Wonder whether it'd be better making it simply use `FixedList` and then add a new field ID validator (in progress PR - https://github.com/hmcts/ccd-data-store-api/pull/1194).

File: repository/src/main/resources/db/changelog/db.changelog-rdm-10041.xml (hunk 1 above)
Comment (RebeccaBaker) [reply repository/src/main/resources/db/changelog/db.changelog-rdm-10041.xml]: good spot 👍  done

File: repository/src/main/resources/db/changelog/db.changelog-rdm-10041.xml (hunk 2 above)
Comment (RebeccaBaker) [reply repository/src/main/resources/db/changelog/db.changelog-rdm-10041.xml]: done 👍

File: repository/src/main/resources/db/changelog/db.changelog-rdm-10041.xml (hunk 3 above)
Comment (RebeccaBaker) [reply repository/src/main/resources/db/changelog/db.changelog-rdm-10041.xml]: @danlysiak Spoke with Jamal agreed we could use FixedList and removed the 40 restriction. Reverted this change out

**Review by danlysiak — APPROVED** [review]

**Pull request closed without merging**